src/usual/MetaPLUsDAOToken.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier:	AGPL-3.0
+pragma solidity ^0.8.9;
+
+import {ERC20Base, ERC20} from "mgv_src/toy/ERC20.sol";
+import {UsualTokenInterface} from "mgv_src/usual/UsualTokenInterface.sol";
+import {PLUsTakerProxy} from "mgv_src/usual/PLUsTakerProxy.sol";
+
+// Meta-PLUsDAO token for use in demo
+contract MetaPLUsDAOToken is ERC20 {
+  UsualTokenInterface public immutable _pLUsDAOToken;
+  PLUsTakerProxy public _pLUsTakerProxy;
+  address public immutable _mangrove;
+  address public _pLUsMgvStrat;
+  address public _admin;
+
+  constructor(
+    address admin,
+    string memory _name,
+    string memory _symbol,
+    UsualTokenInterface pLUsDAOToken,
+    address mangrove
+  ) ERC20Base(_name, _symbol) ERC20(_name) {
+    _pLUsDAOToken = pLUsDAOToken;
+    _mangrove = mangrove;
+    _admin = admin;
+  }
+
+  modifier onlyAdmin() {
+    require(msg.sender == _admin, "MetaPLUsDAOToken/adminOnly");
+    _;
+  }
+
+  function setPLUsTakerProxy(PLUsTakerProxy pLUsTakerProxy) external onlyAdmin {
+    _pLUsTakerProxy = pLUsTakerProxy;
+  }
+
+  function setPLUsMgvStrat(address pLUsMgvStrat) external onlyAdmin {
+    _pLUsMgvStrat = pLUsMgvStrat;
+  }
+
+  function totalSupply() public view virtual override returns (uint) {
+    return _pLUsDAOToken.totalSupply();
+  }
+
+  function balanceOf(address account) public view virtual override returns (uint) {
+    return _pLUsDAOToken.balanceOf(account);
+  }
+
+  function approve(address spender, uint amount) public override returns (bool) {
+    return _pLUsDAOToken.approve(spender, amount) && super.approve(spender, amount);
+  }
+
+  // Only allow the following transfers:
+  //   PLUsMgvStrat -> Mangrove
+  //   Mangrove     -> any address
+  // When owner = Mangrove  =>  transfer & unlock
+  function _transfer(address owner, address recipient, uint amount) internal override returns (bool) {
+    require(
+      (
+        (owner == address(_pLUsMgvStrat) && recipient == _mangrove)
+          || owner == _mangrove && recipient == address(_pLUsTakerProxy)
+      ) && msg.sender == _mangrove,
+      "MetaPLUsDAOToken/nonMangroveTransfer"
+    );
+
+    if (owner == _mangrove) {
+      address currentTaker = _pLUsTakerProxy.currentTaker();
```

Comment:
This requires advanced scrutiny because the `marketOrder` call in `_pLUsTakerProxy` is possibly reentrant due to the bounty transfer.

Reply:
Interesting point. But i do not see an issue in the taker calling `marketOrder` on `_pLUsTakerProxy` again, when receiving the bounty. And it is not possible to do it in a posthook, since the Meta token only allows the transfer from our strat or mgv, therefore it would never call the posthook.